remediation_worker/jobs/kinesis_encrypt_stream/kinesis_encrypt_stream.py

```diff
@@ -0,0 +1,173 @@
+# Copyright (c) 2021 VMware Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import json
+import logging
+import sys
+
+import boto3
+
+logging.basicConfig(level=logging.INFO)
+
+
+class KinesisEncryptStream():
+    def parse(self, payload):
+        """Parse payload received from Remediation Service.
+        Args:
+            payload: (str) JSON string containing parameters
+                    received from the remediation service.
+        Raises:
+            Exception: JSONDecodeError
+        Returns:
+            (dict) Dictionary of parsed parameters.
+        """
+        remediation_entry = json.loads(payload)
+        notification_info = remediation_entry.get("notificationInfo", None)
+        finding_info = notification_info.get("FindingInfo", None)
+
+        stream_name = finding_info.get("ObjectId", None)
+        region = finding_info.get("Region", None)
+
+        if stream_name is None:
+            logging.error("Missing parameters for 'STREAM_NAME'.")
+            raise Exception("Missing parameters for 'stream_name'.")
+
+        if region is None:
+            logging.error("Missing parameters for 'REGION'.")
+            raise Exception("Missing parameters for 'REGION'.")
+
+        logging.info("parsed params:")
+        logging.info("  stream_name: %s", stream_name)
+        logging.info("  region: %s", region)
+
+        return {
+            "stream_name": stream_name,
+            "region": region
+        }
+
+    def remediate(self, stream_name, client, region):
+        """Encrypts unencrypted kinesis data streams.
+        Args:
+            stream_name ([str]): The name of the kinesis stream.
+            client: Instance of the AWS boto3 client.
+        """
+
+        logging.info(
+            "Beginning kinesis data stream encryption evaluation."
+        )
+
+        # Check to see if kinesis data stream is encrypted
+        encryption_type = self.check_encryption(client, stream_name)
+
+        # If kinesis data stream is not encrypted
+        if encryption_type == "NONE":
+            self.enable_encryption(client, stream_name)
+        # If encryption is enabled
```

Comment:
There is no exit or return statement after enabling the encryption. You can write only one return for both if and elif block. i.e write the return statement after if-elif-else block. Something like this -

```
def remediate(self, stream_name, client, region):
        """Encrypts unencrypted kinesis data streams.
        Args:
            stream_name ([str]): The name of the kinesis stream.
            client: Instance of the AWS boto3 client.
        """

        logging.info(
            "Beginning kinesis data stream encryption evaluation."
        )

        # Check to see if kinesis data stream is encrypted
        encryption_type = self.check_encryption(client, stream_name)

        # If kinesis data stream is not encrypted
       if encryption_type == "NONE":
            self.enable_encryption(client, stream_name)
        # If encryption is enabled
        # We have nothing to do.
        # log and exit
        elif encryption_type == "KMS":
            logging.info(
                "Kinesis data stream %s is already encrypted, taking no action.",
                stream_name
            )
        # If anything else returns,
        # log and exit
        else:
            logging.error(
                "An unknown encryption type returned for kinesis data stream %s. Exiting.",
                stream_name
            )
            sys.exit(1)
        return 0
```